docs/source/howto/workflows.rst

```diff
@@ -4,14 +4,217 @@
 How to run multi-step workflows
 *******************************
 
-
 .. _how-to:workflows:write:
 
 Writing workflows
 =================
 
-`#3991`_
+A workflow in AiiDA is a :ref:`process <topics:processes:concepts>` that calls other workflows and calculations and optionally *returns* data and as such can encode the logic of a typical scientific workflow.
+Currently, there are two ways of implementing a workflow process:
+
+ * :ref:`work functions<topics:workflows:concepts:workfunctions>`
+ * :ref:`work chains<topics:workflows:concepts:workchains>`
+
+Here we present a brief introduction on how to write both workflow types.
+
+.. note::
+
+    For more details on the concept of a workflow, and the different between a work function and a work chain, please see the corresponding :ref:`topics section<topics:workflows:concepts>`.
+
+Work function
+-------------
+
+A *work function* is a process function that calls one or more calculation functions and *returns* data that has been *created* by the calculation functions it has called.
+Writing a work function that can be stored in the provenance simply involves writing a Python function that calls a set of calculation functions with the desired logic and decorating it with the ``@workfunction`` decorator:
+
+.. code-block:: python
+
+    from aiida.engine import calcfunction, workfunction
+    from aiida.orm import Int
+
+    @calcfunction
+    def add(x, y):
+        return x + y
+
+    @calcfunction
+    def multiply(x, y):
+        return x * y
+
+    @workfunction
+    def add_and_multiply(x, y, z):
+        sum = add(x, y)
+        product = multiply(sum, z)
+        return product
```

Comment:
```suggestion
        return multiply(sum, z)
```

Reply:
For now I would leave the code examples in the how-to as it is currently implemented in develop. Later we can adjust the code and use `.. include::` to add it directly. I've added this change to the `add_multiply` work function to a local branch where I can still do some small tweaks on the `arithmetic` processes.

Reply:
Ok, sure, then let's leave that. I'm just usually striving to keep the demonstrative examples in the How-Tos as concise as possible.

Reply:
OK, I will just create a ticket on the project to make sure to quickly update this once we have merged docs-revamp into develop

Reply:
See #4128